src/components/modal/__internal__/modal-manager.test.tsx

```diff
@@ -0,0 +1,325 @@
+import ModalManager, { ModalManagerInstance } from "./modal-manager";
+
+let cb1: jest.Mock,
+  cb2: jest.Mock,
+  cb3: jest.Mock,
+  mockModal: HTMLElement | null,
+  mockModal1: HTMLDivElement,
+  mockModal2: HTMLDivElement,
+  mockModal3: HTMLElement | null,
+  otherModal: HTMLElement | null;
+
+test("when the addModal method has been called, then the element passed in an attribute should be the topmost element", () => {
+  cb1 = jest.fn();
+  cb2 = jest.fn();
+
+  mockModal1 = document.createElement("div");
+  mockModal2 = document.createElement("div");
+
+  ModalManager.clearList();
```

Comment:
**comment**: I see that we're calling this `clearList` method in the setup for many of these tests, although not all of them. I assume this is done to clean up the `ModalManager` between tests to stop them interfering with each other - but in this case I would prefer that this is called in an `afterEach` so that it is clear that this is cleanup that happens after each test.

Reply:
I'll get this `afterEach` suggestion implemented 👍🏻